src/services/groups.py

```diff
@@ -115,6 +114,19 @@ async def create_group(self, group_data: GroupCreate) -> GroupResponse:
     async def list_groups(self) -> list[GroupWithScopesResponse]:
         return await self.groups_repository.get_all(self.service_provider_id)
 
+    async def search_groups_by_contract(
```

Comment:
I think the name here can be more explicit, is it a search by contract or by email?